client/src/views/Settings/Tabs/EditProfile/index.vue

```diff
@@ -323,61 +333,85 @@ export default {
+        .then((res) => {
+          this.profile.user_background = res.data.imageUrl
+          this.isNewCoverPhotoUploaded = true
+          this.sendProfile()
+        })
+        .catch(() => {
+          this.loading = false
+        })
+    },
+
     onSubmit () {
+      this.loading = true
+      this.isNewAvatarUploaded = false
+      this.isNewCoverPhotoUploaded = false
+      if (this.isNewAvatarSelected) {
+        this.$refs.avatarEditor.getCroppedData()
+      }
+      if (this.isNewCoverPhotoSelected) {
+        this.$refs.coverPhotoEditor.getCroppedData()
+      }
+      this.sendProfile()
+    },
+
+    sendProfile () {
+      if ((this.isNewAvatarSelected && !this.isNewAvatarUploaded) || (this.isNewCoverPhotoSelected && !this.isNewCoverPhotoUploaded)) {
+        return
+      }
       const profile = { ...this.profile, socialAccounts: [] }
       Object.keys(this.profile.socialAccounts).forEach(item => {
         profile.socialAccounts.push({ type: item, link: this.profile.socialAccounts[item] })
       })
       if (!profile.interests) { profile.interests = [] }
 
-      const arrayPromiseAll = []
-      if (this.profileGetter.avatar_url !== this.profile.avatar_url && this.profile.avatar_url !== "") {
-        const avatarFormData = new FormData()
-        avatarFormData.append('file', this.profile.avatar_url)
-
-        arrayPromiseAll.push(authService.uploadAvatar(avatarFormData, { type: 'avatar' }))
-      }
-      if (this.profileGetter.user_background !== this.profile.user_background && this.profile.user_background !== "") {
-        const coverPhotoFormData = new FormData()
-        coverPhotoFormData.append('file', this.profile.user_background)
-
-        arrayPromiseAll.push(authService.uploadBackground(coverPhotoFormData, { type: 'background' }))
-      }
-      Promise.all(arrayPromiseAll)
-        .then((res) => {
-          console.log(res)
-          if (res.length === 1) {
-            if (this.profileGetter.avatar_url !== this.profile.avatar_url) {
-              profile.avatar_url = res[0].data.imageUrl
-            } else {
-              profile.user_background = res[0].data.imageUrl
-            }
-          }
-          if (res.length === 2) {
-            profile.avatar_url = res[0].data.imageUrl
-            profile.user_background = res[1].data.imageUrl
-          }
-
-          this.$store.dispatch('profile/updateMyProfile', profile)
-            .then(() => {
-              this.$q.notify({
-                ...this.notifyParameters,
-                color: 'primary',
-                message: 'Your profile was edited.'
-              })
-            })
-            .catch(err => {
-              this.$q.notify({
-                ...this.notifyParameters,
-                color: 'negative',
-                message: err && err.response && err.response.data ? err.response.data.error : 'Unknown error.'
-              })
-            })
+      this.$store.dispatch('profile/updateMyProfile', profile)
```

Comment:
After demo please try following the code style used among your team, getters, actions, mutations should be imported via ...mapGetters etc.
Seeing them as imports cleverly indicates what type of changes on the store this component does.